torchaudio/_internal/module_utils.py

```diff
@@ -67,7 +67,10 @@ def wrapped(*args, **kwargs):
 
 
 def is_kaldi_available():
-    return is_module_available("torchaudio._torchaudio") and torch.ops.torchaudio.is_kaldi_available()
+    try:
+        return torch.ops.torchaudio.is_kaldi_available()
+    except Exception:
+        return False
```

Comment:
why treat kaldi differently from sox below? doesn't the requirement of `torchaudio.lib.libtorchaudio` existing hold true here too?

Reply:
The torchaudio codebase is guarded against the lack of C++ extension.
For example, you can clone the repo and do `import torchaudio` on the source directory without building.

So we make these helper function works even when libtorchaudio does not exist.

Reply:
makes sense. then, what's the reasoning for not using a try-catch in `is_sox_available`? seems like the logic in `is_sox_available` should apply here, e.g. we check that `libtorchaudio` exists via `is_module_available`, and if it does, we run `torch.ops.torchaudio.is_kaldi_available`

Reply:
So the difference is the existence of PyBind-based wrapper. Kaldi-feature is implemented in libtorchaudio, which is a C++ library without a link to libPython. So Python does not see it as a Python module, and `is_module_available` does not work. We can check the existence of the `libtorchaudio` file as an added guard statement, but the existence of the file does not necessarily mean that `torch.ops.torchaudio.is_kaldi_available` is available. (also, the extension is different on OS, so it's a bit annoying to do the file existence check) So it's easier to just do `try ~ except` here.

For the case of sox, the implementation will be found in `libtorchaudio_sox` library, and there will be always `torchaudio._torchaudio_sox` wrapper, which is what Python can recognize as a valid Python module (`is_module_available` works). So When `is_module_available("torchaudio.lib._torchaudio_sox")` (changing the target extension name/location to the one I am actually planning to use) succeed, it means that sox-related features are available and `torch.ops.torchaudio.is_sox_available()` exist.